frontend/src/components/user/VendingLink.tsx

```diff
@@ -23,32 +18,44 @@ const VendingLink: FunctionComponent = () => {
 
   // Serial queries needed to get the right vending link
   // First check the user's existing status
-  const statusQuery = useQuery<VendingStatus, string>({
+  const statusQuery = useQuery({
     queryKey: ["/vending/status"],
-    queryFn: getVendingStatus,
+    queryFn: () =>
+      vendingApi.statusVendingStatusGet({
+        withCredentials: true,
+      }),
   })
 
   // Dashboard is accessible as soon as onboarding is submit
-  const dashboardQuery = useQuery<string, string>({
+  const dashboardQuery = useQuery({
     queryKey: ["/vending/status/dashboardlink"],
-    queryFn: getDashboardLink,
-    enabled: statusQuery.isSuccess && statusQuery.data.details_submitted,
-  })
+    queryFn: () =>
+      vendingApi.getDashboardLinkVendingStatusDashboardlinkGet({
+        withCredentials: true,
+      }),
 
-  // Otherwise onboarding required still
-  // Can't pre-fetch link, it will create an account without user action
-  const startOnboard = useCallback(async () => {
-    if (onboarding) return
+    enabled: statusQuery.isSuccess && statusQuery.data.data.details_submitted,
```

Comment:
This is the condition which is causing errors in the frontend (becomes true when it shouldn't), which is interesting because it doesn't happen on `main` even though the logic looks identical apart from the extra `.data` attribute in the chain.

Reply:
Well I guess that was caused by the 201 and us then trying to access `details_submitted`?